File: NeetCode/Arrays-Hashing/Concatenation_of_array.py
# Date: 1 Jan 2025
# Link: https://leetcode.com/problems/concatenation-of-array/
# Name: 1929. Concatenation of Array

from typing import List


class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        return nums+nums


obj = Solution()
nums = [1, 2, 1]
print(obj.getConcatenation(nums))

File: NeetCode/Arrays-Hashing/Concatenation_of_array.cpp
// Date: 28 April 2025
// Link: https://leetcode.com/problems/concatenation-of-array/
// Name: 1929. Concatenation of Array

#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    vector<int> getConcatenation(const vector<int> &nums)
    {
        vector<int> result = nums;
        result.insert(result.end(), nums.begin(), nums.end());
        return result;
    }
};

int main()
{
    int n;
    cin >> n;
    vector<int> nums(n);
    for (int &x : nums)
        cin >> x;

    Solution obj;
    vector<int> ans = obj.getConcatenation(nums);
    for (int x : ans)
        cout << x << " ";

    return 0;
}